app/src/main/java/agency/five/tmdb/ui/MovieRepository.kt

```diff
@@ -0,0 +1,41 @@
+package agency.five.tmdb.ui
+
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+
+interface MovieRepository {
+    suspend fun getPopularMovies() : Flow<List<MovieItem>>
+    suspend fun getStreamingMovies() : Flow<List<MovieItem>>
+    suspend fun getTVMovies() : Flow<List<MovieItem>>
+    suspend fun getOnRentMovies() : Flow<List<MovieItem>>
+    suspend fun getFavoriteMovies() : Flow<MutableList<MovieItem>>
+    suspend fun addFavoriteMovie(movie : MovieItem) : List<MovieItem>
+    suspend fun removeFromFavorites(movie : MovieItem) : List<MovieItem>
+}
+
+internal class MovieRepositoryImpl(private val movieApi : MovieApi) : MovieRepository {
+
+    private val database : Database = Database()
```

Comment:
This should also be injected to the repo through the constructor as your MovieApi